### PR TITLE
Fill global page with all data

### PR DESCRIPTION
**Pull request by JetyAdam**
Built the frontend for global learning resources page. Loaded quickstart data. 

https://issues.redhat.com/browse/RHCLOUD-33582

### CODE REVIEW

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.lr-c-global-learning-resources-page__content {
+  padding: 20px;
+
+  &--gallery {
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.scss:4]: THis should not exist. The PF gallery component takes of of this layout type.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
+.lr-c-global-learning-resources-page__content {
+  padding: 20px;
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.scss:2]: Use PF class name on the element or if you can;t use the  PF variable.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        display: flex;
+        flex-direction: column;
+      }
+    }
+  }
+
+  &--empty {
+    max-width: 441px;
+  }
+}
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.scss:27]: missing new line

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import {
+  Button,
+  EmptyState,
+  EmptyStateActions,
+  EmptyStateBody,
+  EmptyStateFooter,
+  EmptyStateHeader,
+  EmptyStateIcon,
+} from '@patternfly/react-core';
+import CubesIcon from '@patternfly/react-icons/dist/esm/icons/cubes-icon';
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx:16]: imports should be from `/dynamic` not `esm` or `js`

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+import GlobalLearningResourcesQuickstartItem from './GlobalLearningResourcesQuickstartItem';
+import { useSearchParams } from 'react-router-dom';
+
+type Foobar<AsyncFunc extends (...args: any[]) => Promise<unknown>> = (
+  ...args: Parameters<AsyncFunc>
+) => Awaited<ReturnType<AsyncFunc>>;
+
+interface GlobalLearningResourcesContentProps {
+  purgeCache: () => void;
+  loader: Foobar<typeof fetchSuperQuickstarts>;
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx:27]: I think this should be imported from the FEC utils suspense loader. `UnwrappedLoader`

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+const GalleryQuickstart: React.FC<GalleryQuickstartProps> = ({
+  quickStarts,
+  purgeCache,
+}) => {
+  return (
+    <Gallery className="lr-c-global-learning-resources-page__content--gallery">
+      {quickStarts.map((quickStart) => {
+        return (
+          <div className="lr-c-global-learning-resources-page__content--gallery-card-wrapper">
+          <GlobalLearningResourcesQuickstartItem
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx:44]: I think linter will yell at this line.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+            icon={<EmptyStateIcon icon={CubesIcon} />}
+          />
+          <EmptyStateBody>
+            You don&apos;t have any bookmarked learning resources. Click the
+            icon in cards on the &apos;All learning resources&apos; tab to
+            bookmark a resource.
+          </EmptyStateBody>
+          <EmptyStateFooter>
+            <EmptyStateActions>
+              <Button variant="link">
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx:80]: I don't think this link leads to anywhere.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+              </Button>
+            </EmptyStateActions>
+          </EmptyStateFooter>
+        </EmptyState>
+      </Bullseye>
+    );
+  }
+  return (
+    <Gallery className="lr-c-global-learning-resources-page__content--gallery">
+      {quickStarts.map((quickStart) => {
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx:91]: These should be first filtered so we are not adding `null` nodes to the tree.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+};
+
+const GlobalLearningResourcesContent: React.FC<
+  GlobalLearningResourcesContentProps
+> = ({ loader, purgeCache }) => {
+  const chrome = useChrome();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  useEffect(() => {
+    setSearchParams({ tab: 'all' });
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesContent.tsx:115]: I'd make an enum from this.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.scss [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    display: flex;
+    align-items: center;
+  
+    &-icon {
+      width: 16px;
+      height: 16px;
+      margin-right: 5px;
+    }
+  }
+}
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.scss:16]: Missing new line

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.scss [hunk 11]
```diff
@@ -0,0 +1,16 @@
+.lr-c-global-learning-resources-page__filters {
+  box-sizing: border-box;
+  padding: 20px;
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.scss:3]: Use Spacing classname or variable.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.scss [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  background-color: var(--pf-v5-global--palette--black-100);
+
+  &--checkbox {
+    display: flex;
+    align-items: center;
+  
+    &-icon {
+      width: 16px;
+      height: 16px;
+      margin-right: 5px;
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.scss:13]: Spacing variable

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.tsx [hunk 13]
```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import { InputGroup, InputGroupItem, InputGroupText, Flex, FlexItem, Text, TextInput, TextVariants, TextContent, TextInputGroup, TextInputGroupMain, TextInputGroupUtilities, Stack, StackItem, Checkbox, Button, Split, SplitItem } from '@patternfly/react-core';
+import { AngleDownIcon, FilterIcon, SortAmountDownAltIcon } from '@patternfly/react-icons';
+import { right } from '@patternfly/react-core/dist/esm/helpers/Popper/thirdparty/popper-core';
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.tsx:4]: I would not use this import at all. Or use dynamic.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.tsx [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+  group: string;
+  data: string[];
+}
+
+interface Category {
+  category: string;
+  data: SubCategory[];
+}
+
+const data = [
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.tsx:18]: A NOTE: just temp data before API is done.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.tsx [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+      ))}
+    </Stack> 
+  )};
+
+  export default GlobalLearningResourcesFilters;
+
+  
+
+
+
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFilters.tsx:76]: Many extra lines.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFiltersCategory.tsx [hunk 16]
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import { Checkbox, ExpandableSection, Stack, StackItem, Text, TextVariants, TextContent } from '@patternfly/react-core';
+
+interface SubCategory {
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFiltersCategory.tsx:4]: I think I saw this type before, extract it and share the definition.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFiltersCategory.tsx [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+  group: string;
+  data: string[];
+}
+
+interface CategoryProps {
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesFiltersCategory.tsx:9]: ^^

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesHeader.tsx [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+import './GlobalLearningResourcesHeader.scss';
+import { Divider, Stack, StackItem, Text, TextContent, TextVariants } from '@patternfly/react-core';
+
+export const GlobalLearningResourcesHeader = () => {
+  return (
+    <div className='lr-c-global-learning-resources-page__header'>
+        <img className='pf-v5-u-ml-lg pf-v5-u-mr-lg' width="48px" height="48px" src="/apps/frontend-assets/console-landing/ansible.svg" alt="Learning resources icon" />
+        <Divider
+        className='pf-v5-u-pt-md pf-v5-u-pb-md'
+        height=""
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesHeader.tsx:11]: empty prop?

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesHeader.tsx [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+          default: 'insetSm',
+        }}
+      />
+        <TextContent>
+        <Stack className='pf-v5-u-m-lg'>
+          <StackItem>
+            <Text component={TextVariants.h1}>All Learning Resources</Text>
+          </StackItem>
+          <StackItem>
+            <Text component={TextVariants.p}>See learning resources for services and features across the Hybrid Cloud Console. Find additional resources on <Text component={TextVariants.a} href="https://developers.redhat.com/">Developers.RedHat.com</Text>, <Text component={TextVariants.a} href="https://cloud.redhat.com/">Cloud.RedHat.com</Text>, and on <Text component={TextVariants.a} href="https://www.redhat.com/">RedHat.com</Text>.</Text>
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesHeader.tsx:25]: I think these links should open new tab.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.scss [hunk 20]
```diff
@@ -0,0 +1,13 @@
+.lr-c-global-learning-resources-page {
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+
+  &__main {
+    display: grid;
+    grid-template-columns: 22% 78%;
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.scss:8]: Follow UP, work with UX and get mocks for various screen sizes.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.scss [hunk 21]
```diff
@@ -0,0 +1,13 @@
...
+    flex: 1;
+    height: 100%;
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.scss:10]: What happens if the window is short? Do we get scrollbar?

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.tsx [hunk 22]
```diff
@@ -1,7 +1,35 @@
...
+    <div className="lr-c-global-learning-resources-page">
+      <div className="lr-c-global-learning-resources-page__top-container">
+        <GlobalLearningResourcesHeader />
+        <Suspense fallback={<GlobalLearningResourcesTabs />}>
+          <GlobalLearningResourcesTabs loader={loader} />
+        </Suspense>
+      </div>
+      <div className="lr-c-global-learning-resources-page__main">
+        <GlobalLearningResourcesFilters />
+        <Suspense fallback={<Spinner diameter="80px" aria-label="Learning resources are being loaded." />}>
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.tsx:24]: Just check if we can use the predefined sizes ("xl" maybe?)

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesQuickstartItem.tsx [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+interface GlobalLearningResourcesQuickstartItemProps {
+  quickStart: QuickStart;
+  purgeCache: () => void;
+}
+
+const GlobalLearningResourcesQuickstartItem: React.FC<
+  GlobalLearningResourcesQuickstartItemProps
+> = ({ quickStart, purgeCache }) => {
+  const chrome = useChrome();
+  const showBookmarks = useFlag('platform.learning-resources.bookmarks');
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesQuickstartItem.tsx:32]: This should be always true. I think we can remove it.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.scss [hunk 24]
```diff
@@ -0,0 +1,14 @@
+.lr-c-global-learning-resources-tabs {
+  padding: 10px 0 0 20px;
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.scss:2]: Use spacing classes.

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.tsx [hunk 25]
```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { Spinner, Tab, TabTitleText, Tabs } from '@patternfly/react-core';
+import './GlobalLearningResourcesTabs.scss';
+import fetchSuperQuickstarts from '../../utils/fetchQuickstarts';
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import { QuickStart } from '@patternfly/quickstarts';
+import { Link } from 'react-router-dom';
+import { useSearchParams } from 'react-router-dom';
+
+type Foobar<AsyncFunc extends (...args: any[]) => Promise<unknown>> = (
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.tsx:10]: FOOBAR here ☎️

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.tsx [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+  ...args: Parameters<AsyncFunc>
+) => Awaited<ReturnType<AsyncFunc>>;
+
+interface GlobalLearningResourcesTabsProps {
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.tsx:14]: I think I saw this type before. Share it?

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.tsx [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+      activeKey={searchParams.get('tab')!}
+    >
+      <Tab
+        eventKey="all"
+        title={
+          <Link
+            className="lr-c-global-learning-resources-tabs__link"
+            to={{
+              pathname: '.',
+              search: `?tab=all`,
```
Comment (Hyperkid123) [inline src/components/GlobalLearningResourcesPage/GlobalLearningResourcesTabs.tsx:44]: We should get the `all` and `bookmarks` from some TS enum.

File: src/utils/LoaderReaderQuickStarts.tsx [hunk 28]
```diff
@@ -0,0 +1,20 @@
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import { ChromeAPI } from '@redhat-cloud-services/types';
+import React from 'react';
+
+interface LoaderReaderQuickStartsProps {
+  loader: (chrome: ChromeAPI) => string[];
+}
+
+const LoaderReaderQuickStarts: React.FC<LoaderReaderQuickStartsProps> = ({
```
Comment (Hyperkid123) [inline src/utils/LoaderReaderQuickStarts.tsx:9]: Is this used anywhere?

File: src/utils/fetchQuickstarts.ts [hunk 29]
```diff
@@ -0,0 +1,60 @@
+import { ChromeAPI } from '@redhat-cloud-services/types';
+import { API_BASE, FAVORITES, FavoriteQuickStart } from './toggleFavorite';
+import { QUICKSTARTS } from '../hooks/useQuickStarts';
+import axios from 'axios';
+import { QuickStart } from '@patternfly/quickstarts';
+
+async function fetchSuperQuickstarts(
```
Comment (Hyperkid123) [inline src/utils/fetchQuickstarts.ts:7]: This probably deserves some other name. 🥇

File: src/utils/fetchQuickstarts.ts [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+  const account = user.identity.internal?.account_id;
+
+  const quickstartsPath = `${API_BASE}/${QUICKSTARTS}`;
+
+  const contentPromise = axios
+    .get<{ data: { content: QuickStart }[] }>(quickstartsPath, {
+      params: { account, bundle },
+    })
+    .then(({ data }) => {
+      // `${account}`,
```
Comment (Hyperkid123) [inline src/utils/fetchQuickstarts.ts:25]: comment

File: src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.scss (hunk 20 above)
Comment (JetyAdam) [reply src/components/GlobalLearningResourcesPage/GlobalLearningResourcesPage.scss:8]: Talking to UX atm

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**